### PR TITLE
Fix test timing issue + minor code cleanup

### PR DESCRIPTION
**Pull request by narrieta**
The agent_firewall test may attempt to check the firewall before the Agent has set it up. For iptables, this was not an issue because the firewall is setup by the daemon, but that is not the case for nft. 

Added an extra check to verify that the walinuxagent table is already created in NfTables.get_missing_rules().

The Agent may still be in the middle of setting up the table when the test runs. My next PR will add atomic updated for nft, which should take care of that condition.

The PR also includes some minor cleanup that I pointed out as comments.

### CODE REVIEW

File: azurelinuxagent/ga/firewall_manager.py [hunk 1]
```diff
@@ -72,20 +72,27 @@ def create(wire_server_address):
         """
         try:
             manager = IpTables(wire_server_address)
-            logger.info("Using iptables to manage firewall rules")
+            event.info(WALAEventOperation.Firewall, "Using iptables [version {0}] to manage firewall rules", manager.version)
```
Comment (narrieta) [inline azurelinuxagent/ga/firewall_manager.py:75]: added telemetry to get the version of iptables/nft/firewall-cmd

File: tests/ga/test_env.py [hunk 2]
```diff
@@ -169,7 +169,8 @@ def test_it_should_restore_missing_firewall_rules(self):
             test_cases = [  # Exit codes for the "-C" (check) command
                 {"accept_dns": 1, "accept": 0, "drop": 0, "legacy": 0},
                 {"accept_dns": 0, "accept": 1, "drop": 0, "legacy": 0},
-                {"accept_dns": 0, "accept": 0, "drop": 1, "legacy": 0},
+                {"accept_dns": 0, "accept": 1, "drop": 0, "legacy": 0},
+                {"accept_dns": 1, "accept": 1, "drop": 1, "legacy": 0},
```
Comment (narrieta) [inline tests/ga/test_env.py:173]: added test case for all rules missing

File: tests_e2e/tests/agent_firewall/agent_firewall.py [hunk 3]
```diff
@@ -33,9 +32,7 @@ def __init__(self, context: AgentVmTestContext):
         self._ssh_client = self._context.create_ssh_client()
 
     def run(self):
-        log.info("Checking firewall rules added by the agent")
```
Comment (narrieta) [inline tests_e2e/tests/agent_firewall/agent_firewall.py:36]: Removed those messages; they are sort of confusing

File: tests_e2e/tests/lib/agent_log.py [hunk 4]
```diff
@@ -369,18 +369,6 @@ def get_errors(self) -> List[AgentLogRecord]:
                 'if': lambda r: r.prefix == 'ExtHandler' and r.thread == 'ExtHandler'
             },
             #
-            # TODO: Currently Ubuntu minimal does not include the 'iptables' command. Remove this rule once this has been addressed.
```
Comment (narrieta) [inline tests_e2e/tests/lib/agent_log.py:372]: no longer needed

File: tests_e2e/tests/lib/firewall_manager.py [hunk 5]
```diff
@@ -264,17 +264,19 @@ def get_state(self) -> str:
     }
 
     def get_missing_rules(self) -> List[str]:
-        missing = []
+        if "table ip walinuxagent" not in shellutil.run_command(["sudo", "nft", "list", "tables"]):
```
Comment (narrieta) [inline tests_e2e/tests/lib/firewall_manager.py:267]: add check for case when the table has not been created yet

**Review by nagworld9 — APPROVED** [review]

**Review by maddieford — APPROVED** [review]

**Pull request merged**